bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java

```diff
@@ -0,0 +1,160 @@
+    private HttpClient httpClient;
+
+    public TeslaPowerwallWebTargets(String hostname, HttpClient httpClient) {
+        this.httpClient = httpClient;
+        String baseUri = "https://" + hostname + "/";
+        getBatterySOEUri = baseUri + "api/system_status/soe";
+        getGridStatusUri = baseUri + "api/system_status/grid_status";
+        getSystemStatusUri = baseUri + "api/system_status";
+        getMeterAggregatesUri = baseUri + "api/meters/aggregates";
+        getTokenUri = baseUri + "api/login/Basic";
+        getOperationUri = baseUri + "api/operation";
+    }
+
+    public BatterySOE getBatterySOE(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getBatterySOEUri, email, password);
+        logger.trace("getBatterySOE response = {}", response);
+        return BatterySOE.parse(response);
+    }
+
+    public GridStatus getGridStatus(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getGridStatusUri, email, password);
+        logger.trace("getGridStatus response = {}", response);
+        return GridStatus.parse(response);
+    }
+
+    public SystemStatus getSystemStatus(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getSystemStatusUri, email, password);
+        logger.trace("getSystemStatus response = {}", response);
+        return SystemStatus.parse(response);
+    }
+
+    public MeterAggregates getMeterAggregates(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getMeterAggregatesUri, email, password);
+        logger.trace("getMeterAggregates response = {}", response);
+        return MeterAggregates.parse(response);
+    }
+
+    public Operations getOperations(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getOperationUri, email, password);
+        logger.trace("getOperations response = {}", response);
+        return Operations.parse(response);
+    }
+
+    public String getToken(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("username", "customer");
+        jsonObject.addProperty("password", password);
+        jsonObject.addProperty("email", email);
+        jsonObject.addProperty("force_sm_off", false);
+        logger.debug("logonjson = {}", jsonObject.toString());
+        String response = invoke(getTokenUri, HttpMethod.POST.asString(), "Content-Type", "application/json",
```

Comment:
```suggestion
        String response = invoke(getTokenUri, HttpMethod.POST, "Content-Type", "application/json",
```

This should be coombined with the opther comment about the method signature.

Reply:
Done.